### ISSUE TITLE
[Feature Request] Thumbnails in Power Lora Loader

### ISSUE THREAD

**Issue by Davikar** [rgthree/rgthree-comfy#223]
The Power Lora Loader is great, but there's one key feature that is missing that's keeping me from using it everywhere...
It would be nice if we could have thumbnails and lora info like there is in Pythongosssss lora loader, but for each lora.
Now, I don't know how easy it would be to implement, but throwing out this suggestion here in case it's doable.

**Comment by JorgeR81** [comment]
I vote for this ! 
If you use the Power Lora Loader, you have many loras, and it's easier to lose track of their names. 

The Pythongosssss lora loader creates 2 files, per lora, in the models folder.
One is for the image thumbnail, and another for extra data.
You can add the thumbnail only on the loras you want ( and you can use the image from Civitai, directly ).
You can also add your own notes, to the lora info !

I'm assuming that everyone that wants this feature already has the Pythongosssss nodes. It's one of the main features. 
But the Pythongosssss lora loader, it's for a single lora, and it's even larger than the default loader. 

Maybe you could just add a way to read these 2 files, via the Lora Power Loader.
This way the implementation would be easier and we avoid duplicated features / files.

![loras](https://github.com/rgthree/rgthree-comfy/assets/72734038/73d3e651-60fd-42a8-8619-a67e7a4efc4a)

![pg2](https://github.com/rgthree/rgthree-comfy/assets/72734038/fea065ae-a50d-409e-b4cb-a736308b2a0d)

![pg3](https://github.com/rgthree/rgthree-comfy/assets/72734038/6f1d3af8-4fc5-466a-8865-9e9985eca0d6)

![pg1](https://github.com/rgthree/rgthree-comfy/assets/72734038/8820fc14-efea-4d47-ab5f-445d7322d70f)

![pg3](https://github.com/rgthree/rgthree-comfy/assets/72734038/65ca9db8-bb13-4a7f-9c9e-c12cdb640f0a)

**Comment by rgthree** [comment]
The Power Lora Loader has some new changes coming soon that you folks will probably love :)

**Comment by murphylanga** [comment]
I am looking forward to it. If you work on it, maybe it will also be possible to output the Lora names and put them together.

The SD Prompt Saver from the SD Prompt Reader Nodes works very well with your nodes. The Lora model names can be stored in the image.

**Comment by Davikar** [comment]
Okay, just checked out the new changes and it's a nice implementation, but it's lacking some things that the pythongoss loaders have.

<img width="954" alt="image" src="https://github.com/rgthree/rgthree-comfy/assets/8229634/72b55f31-7ee5-4aa1-8dcd-7e8675fe390c">

vs

<img width="607" alt="image" src="https://github.com/rgthree/rgthree-comfy/assets/8229634/ab102a4f-e7ec-4dcd-a9ae-a51571bd139b">

Pythongoss version reads tags from the safetensor files, which are usually a lot more than what's listed on the civitai page.
It also lists the base model and clip skip used for training the loras.

**Comment by rgthree** [comment]
Yes, the new dialog is also not the final version. Though, I was curious about the tags from the metadata. Do people actually find that useful? It's so exhaustive, I usually just trust what is outlined on civitai.

Anyway, this isn't the final dialog or power lora loader experience (and why I didn't post in here the last push :)

**Comment by Davikar** [comment]
Okay, well, the information on Civitai can be limited some times, the creators don't always write everything down.
And a trigger word is just the topmost tag used in every training image, other tags are relevant too and allows you to use the lora better. Plus it's quicker to fire up the info dialog rather than having to load a civitai page just look something up.
Idk, maybe I'm one of the few who looks at them, but it's a nice thing to have.

**Comment by Davikar** [comment]
It's pretty close to perfect now, only thing that is missing now is preview images in the lora list.

<img width="481" alt="image" src="https://github.com/rgthree/rgthree-comfy/assets/8229634/a90b9ada-8fa9-46bf-a86a-eb6378cef505">

**Comment by Piscabo** [comment]
Just a quick question. With my one PC setup, I can see the thumbnails as you scroll through the Lora list. My other setup has no thumbnails. Do I have to change a setting somewhere? They are both the newest. I update with manager daily.

**Comment by Davikar** [comment]
> Just a quick question. With my one PC local ComfyUI setup, I can see the thumbnails as you scroll through the Lora list. My other setup has no thumbnails. Do I have to change a setting somewhere? They are both the newest. I update with manager daily.

The thumbnails are only available in the pythongoss lora loader right now.

**Comment by Piscabo** [comment]
> > Just a quick question. With my one PC local ComfyUI setup, I can see the thumbnails as you scroll through the Lora list. My other setup has no thumbnails. Do I have to change a setting somewhere? They are both the newest. I update with manager daily.
> 
> The thumbnails are only available in the pythongoss lora loader right now.

Then I have a magic node, but only on my one machine.
![image](https://github.com/rgthree/rgthree-comfy/assets/123719842/239bdc45-1588-45c5-93af-9e96c236a889)

**Comment by rgthree** [comment]
The pythongoss script may be hijacking the menu in one but not the other. Not sure why one case would, though.

I’ll see about adding it to the menu manually. I’m also working out a full-fledged Lora adding experience too.

**Comment by JorgeR81** [comment]
In my case, the Power Lora Loader (rgthree) is displaying duplicate images, meant for the pythongosssss lora loader.

The image on the right, with the generation data, is the one fetched the Power Lora Loader (rgthree).

The one left is a jpeg. It was created by pythongosssss lora loader, in the lora folder ( with the same name as the lora ).
This image is used by pythongosssss lora loader to display a preview, as we mouseover the lora list ( see posts above ).
But now it's also being displayed by the Power Lora Loader (rgthree).

The Power Lora Loader (rgthree) keeps displaying the duplicated image, even if the jpeg it's deleted on the lora folder. 
Is it possible to hide these duplicated jpeg images in the Power Lora Loader (rgthree) ?

![img](https://github.com/rgthree/rgthree-comfy/assets/72734038/f67a4d8a-d18b-408a-a74e-0aea7e1668d8)

**Comment by HaydenReeve** [comment]
The Power Lora Loader is my favourite node ever, as it just cleans up your workflow so much. I do agree that the ability to preview an image in the file selector is such an incredibly useful feature that is missing from the current implementation.

Off topic, but the only other thing on my wishlist would be keep the search bar searching the entire directory recursively.

**Comment by Dexintux** [comment]
Hi is there any workaround to make it work?
